### PR TITLE
STRIPES-708: Discourage placeholders in I18N documentation

### PR DESCRIPTION
**Pull request by julianladisch**
I've added these sections:
* Don't split sentences
* Use placeholders for dynamic content
* Avoid placeholders

I've removed the example section because it shows discouraged placeholders use.

https://issues.folio.org/browse/STRIPES-748 "Add example section to doc/i18n.md" exists. Since October 2020 I've requested that a Stripes developer fixes the example section but no one was able to do this task. This justifies to remove the example section until someone picks up [STRIPES-748](https://issues.folio.org/browse/STRIPES-748).

### CODE REVIEW

**Review by zburke — APPROVED** [review]
Thanks for this! 

It's really unfortunate that I chose `ControlledVocab` as an example in this documentation because the way it was written precludes correct i18n 😢 .

File: doc/i18n.md [hunk 1]
```diff
@@ -160,61 +163,126 @@ MyComponent.propTypes = {
...
-export default injectIntl(ControlledVocab);
+A possible German translation is
+
+```
+"idWasFound": "Es wurde {value} gefunden"
+```
+
+## Avoid placeholders
+
+In the above section, placeholders were used for identifiers. They can also safely be used for things like names, or titles, etc. However, avoid placeholders the placeholder is known at compile time and especially if the placeholder can be different nouns or different verbs.
```
Comment (MikeTaylor) [inline doc/i18n.md:233]: "avoid placeholders the placeholder" must be a typo, but it's not obvious what for.

File: doc/i18n.md [hunk 2]
```diff
@@ -160,61 +163,126 @@ MyComponent.propTypes = {
...
+```
 
-export default injectIntl(ControlledVocab);
+A possible German translation is
+
+```
+"idWasFound": "Es wurde {value} gefunden"
+```
+
+## Avoid placeholders
```
Comment (MikeTaylor) [inline doc/i18n.md:231]: This heading is a misleading summary of the section. I would recommend something like "Be cautious when using placeholders" or "avoid substituting nouns and verbs or placeholders".

File: doc/i18n.md [hunk 3]
```diff
@@ -160,61 +163,126 @@ MyComponent.propTypes = {
...
+
+```
+<FormattedMessage id={`ui-feeds.wasFound.${type}`} />
+```
+
+```
+<FormattedMessage id={`ui-feeds.delete.${type}`} />
+```
+
+If there are pre-defined and custom types handle them separately:
```
Comment (MikeTaylor) [inline doc/i18n.md:268]: "If there are both pre-defined ..." would be clearer.

**Review by MikeTaylor — CHANGES_REQUESTED** [review]
Thanks for doing this. See minor comments.

File: doc/i18n.md (hunk 1 above)
Comment (julianladisch) [reply doc/i18n.md]: Resolved.

File: doc/i18n.md (hunk 2 above)
Comment (julianladisch) [reply doc/i18n.md]: Resolved.

File: doc/i18n.md (hunk 3 above)
Comment (julianladisch) [reply doc/i18n.md]: Changed as suggested.

File: doc/i18n.md [hunk 4]
```diff
@@ -160,61 +163,126 @@ MyComponent.propTypes = {
...
+```
 
-export default injectIntl(ControlledVocab);
+A possible German translation is
+
+```
+"idWasFound": "Es wurde {value} gefunden"
+```
+
+## Avoid substituting nouns and verbs or placeholders
```
Comment (MikeTaylor) [inline doc/i18n.md:231]: or -> for. Sorry :-)

File: doc/i18n.md (hunk 1 above)
Comment (MikeTaylor) [reply doc/i18n.md]: Confirmed.

File: doc/i18n.md (hunk 2 above)
Comment (MikeTaylor) [reply doc/i18n.md]: You copied the typo in my suggested title :-)

File: doc/i18n.md (hunk 3 above)
Comment (MikeTaylor) [reply doc/i18n.md]: Confirmed.

**Review by MikeTaylor — APPROVED** [review]
Thanks!

**Pull request merged**